Makefile

```diff
@@ -47,40 +42,64 @@ devel: test
 
 .PHONY: package
 package: test
-	cabal check
-	# tests that a source-distribution can be generated
-	cabal sdist
+	# check and bundle the package
+	stack sdist
+	# outputs to .stack-work/install/<machine-architecture>/<snapshot>/doc
+
 	# check that the generated source-distribution can be built & installed
+	# TODO: where to put build?
 	{ \
-	set -e; \
-	export SRC_TGZ=$$(cabal info . | awk '{print $$2 ".tar.gz";exit}') ; \
-	cd dist/; \
-	cabal sandbox init; \
-	if [ -f "$$SRC_TGZ" ]; then \
-		cabal install --force-reinstalls "$$SRC_TGZ"; \
-	else \
-		echo "expected '$$SRC_TGZ' not found"; \
-		exit 1; \
-	fi; \
+	set -e; set -x; \
+	export SRC_TGZ=$$(stack sdist 2>&1 | tail -n 1) ; \
+	export PACKAGE=$${SRC_TGZ%.tar.gz}; \
+	export PACKAGE=$${PACKAGE##*/}; \
+	mkdir -p .build/; cd .build/; \
+	cp $$SRC_TGZ .; \
+	tar xf $$SRC_TGZ; \
+	cd $$PACKAGE; \
+	pwd; \
+	stack init; \
+	stack build; \
+	stack test; \
+	cd ../..; \
+	rm -r .build; \
 	}
 
+
 .PHONY: test-all
 test-all: build package
 
 .PHONY: ci-test
 ci-test:
-	# run tests for all ghc versions
-	# TODO: figure out a way to run even if in nix-shell
+	# run tests for all ghc versions given in different ghc-versions
 	{ \
 	set -e; \
-	if [ "$$IN_NIX_SHELL" ]; then \
-		echo "Invoked from within nix-shell; exiting"; \
-		exit 1; \
-	else \
-		for e in $(GHCS); do \
-			echo "Testing GHC $$e"; \
-			nix-shell shell-$$e.nix --command "make test-all"; \
-			echo "DONE GHC $$e"; \
-		done; \
-	fi; \
+	for ghc in $(GHCS); do \
+		echo "Testing GHC verison $$ghc"; \
+		stack --stack-yaml="stack/stack-$$ghc.yaml" clean; \
+		stack --stack-yaml="stack/stack-$$ghc.yaml" build; \
+		stack --stack-yaml="stack/stack-$$ghc.yaml" test; \
+	done; \
 	}
+
+.PHONY: stack-clean
+stack-clean:
```

Comment:
@power-fungus `clean` as a target would be simpler. i'm not sure we need to specify it's `stack` based

Reply:
@suhailshergill as noted below, `stack clean` does not enough "cleaning" to call it just `clean`. As I understand it, the intent of the `clean` target is to get rid of all generated files inside the project. For that, `.stack-work`, `dist`, sandbox-stuff had to be removed. I have no complete list of files that need to be deleted and therefore, I did not make a `clean` target.